SQL-Solutions.txt

```diff
@@ -0,0 +1,2743 @@
+ORDER BY o.snum;
+
+Output - 
+
++------+------------+----------------------+------+---------+
+| onum | odate      | largest_order_in_amt | snum | sname   |
++------+------------+----------------------+------+---------+
+| 3003 | 1990-10-03 |               767.19 | 1001 | Peel    |
+| 3008 | 1990-10-05 |                 4723 | 1001 | Peel    |
+| 3011 | 1990-10-06 |              9891.88 | 1001 | Peel    |
+| 3005 | 1990-10-03 |              5160.45 | 1002 | Serres  |
+| 3007 | 1990-10-04 |                75.75 | 1002 | Serres  |
+| 3010 | 1990-10-06 |              1309.95 | 1002 | Serres  |
+| 3009 | 1990-10-04 |              1713.23 | 1003 | AxelRod |
+| 3002 | 1990-10-03 |               1900.1 | 1004 | Motika  |
+| 3001 | 1990-10-03 |              1098.16 | 1007 | Rifkin  |
++------+------------+----------------------+------+---------+
+
+-- 4) Arrange the Order table by descending customer number. 
+
+Query - 
+
+SELECT 
+    *
+FROM
+    orders
+ORDER BY cnum DESC;
+
+Output - 
+
++------+---------+------------+------+------+
+| ONUM | AMT     | ODATE      | CNUM | SNUM |
++------+---------+------------+------+------+
+| 3001 |   18.69 | 1990-10-03 | 2008 | 1007 |
+| 3006 | 1098.16 | 1990-10-03 | 2008 | 1007 |
+| 3002 |  1900.1 | 1990-10-03 | 2007 | 1004 |
+| 3008 |    4723 | 1990-10-05 | 2006 | 1001 |
+| 3011 | 9891.88 | 1990-10-06 | 2006 | 1001 |
+| 3007 |   75.75 | 1990-10-04 | 2004 | 1002 |
+| 3010 | 1309.95 | 1990-10-06 | 2004 | 1002 |
+| 3005 | 5160.45 | 1990-10-03 | 2003 | 1002 |
+| 3009 | 1713.23 | 1990-10-04 | 2002 | 1003 |
+| 3003 |  767.19 | 1990-10-03 | 2001 | 1001 |
++------+---------+------------+------+------+
+
+-- 5) Find which salespeople currently have orders in the order table. 
+
+Query - 
+
+SELECT 
+    *
+FROM
+    salespeople
+WHERE
+    snum IN (SELECT 
+            snum
+        FROM
+            orders);
+
+Output - 
+
++------+---------+-----------+------+
+| SNUM | SNAME   | CITY      | COMM |
++------+---------+-----------+------+
+| 1001 | Peel    | London    |   12 |
+| 1002 | Serres  | SanJose   |   13 |
+| 1003 | AxelRod | New York  |   10 |
+| 1004 | Motika  | London    |   11 |
+| 1007 | Rifkin  | Barcelona |   15 |
++------+---------+-----------+------+
+
+-- 6) List names of all customers matched with the salespeople serving them. 
+
+Query -
+
+SELECT 
+    c.cname AS CUSTOMER_NAME, s.sname AS SERVING_BY
+FROM
+    customers AS c,
+    salespeople AS s
+WHERE
+    s.snum = c.snum;
+
+Output -
+
++---------------+------------+
+| CUSTOMER_NAME | SERVING_BY |
++---------------+------------+
+| Hoffman       | Peel       |
+| Giovanni      | AxelRod    |
+| Liu           | Serres     |
+| Grass         | Serres     |
+| Clemens       | Peel       |
+| Pereira       | Motika     |
+| Cisneros      | Rifkin     |
++---------------+------------+
+
+-- 7) Find the names and numbers of all salespeople who have more than one customer.
+
+Query - 
+
+SELECT 
+    s.sname, s.snum
+FROM
+    salespeople AS s,
```

Comment:
Why Cross Join It is a efficient way?

Reply:
Sir I have used Inner Join not cross Join .
Yes sir it is efficient.
For modern RDBMS there is no difference between "explicit JOIN" and "JOIN-in-the-WHERE" (if all JOINS are INNER) regards performance and query plan.

Reply:
Perforamance b/w where filter and by using on filter there is great performance diffrence. you should have to thougth and explore about it.

Reply:
ok sir